snapcraft/plugins/catkin.py

```diff
@@ -127,77 +160,59 @@ def gcc_version(self):
     @property
     def rosdir(self):
         return os.path.join(self.installdir,
-                            'opt', 'ros', self.options.rosversion)
+                            'opt', 'ros', self.options.rosdistro)
 
-    def _deps_from_packagesxml(self, f, pkg):
-        tree = lxml.etree.parse(f)
+    def _find_dependencies(self):
```

Comment:
I would find this clearer if it is a get function that returns the dependencies, instead of a method with no return value that has a side effect of changing the dependencies attribute. Would that be hard to change?